### PR TITLE
Project Styling Template (Members)

### PR DESCRIPTION
**Pull request by ganmany5288**
I think this is the right place lol

Made a template for members, social medias, and the member's image that can be used across different pages in the website (courseup, vikelabs, ...)

### CODE REVIEW

File: src/templates/project/project-members.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import React from 'react';
+import "./project-styling.css"
+import { Memberdetail, Socials } from "./project-styling"
+import { LiteYoutubeEmbed } from 'react-lite-yt-embed'; // incase it's needed
+import ReactTooltip from 'react-tooltip';
+import Media from './member-media';
+
+const Member = (props) => {
+    return (
+        <Memberdetail>
```
Comment (robchendev) [inline src/templates/project/project-members.js]: Nitpicking: In react components, It's a good practice to make the capital of each word uppercase. This way it's easier to read so we don't run into something like `<Vikelabsgatsbywebsiteproject />`

P.S. for variables, it's good practice to name them the same way but have the first word all lowercase. ex: `vikelabsGatsbyWebsiteProject`

File: src/templates/project/project-styling.js [hunk 2]
```diff
@@ -59,4 +59,87 @@ export const Explanation = styled.div`
...
+  padding: ${v.mdSpacing};
+  border-radius: calc(${v.borderRadius} * 2);
+  
+  display: flex;
+  justify-content: space-between;
+  align-items: center;
+  h1 {
+    font-size: 1.4em;
+    margin-top: -0.2em;
+    margin-bottom: 0.1em;
```
Comment (robchendev) [inline src/templates/project/project-styling.js]: We can shorten this to `margin: -0.2em 0 0.1em 0;`, with the order corresponding to top, right, bottom, left. 

`margin: -0.2em 0 0.1em;` also works, but can be confusing to read.

File: src/templates/project/member-media.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+import { HeadingContainer, PageHeader, EntireWrapper, Explanation, Teststuff, Teammates, Memberdetail, MemberIcon, Socials } from "./project-styling"
+import { LiteYoutubeEmbed } from 'react-lite-yt-embed'; // incase it's needed
+import { FaGithub, FaLinkedin} from 'react-icons/fa'; //Import more if required
+import ReactTooltip from 'react-tooltip';
+import lechonk from '../../images/chonk.png'
+import { IconContext } from 'react-icons';
+import ReactDOM from 'react-dom';
+
+
+const MediaCheck = (props) => {
```
Comment (robchendev) [inline src/templates/project/member-media.js]: Instead of receiving `props` in (), destructure them first so you know exactly what you are receiving as props. It's also easier since you don't have to add `props.` before each prop you use. You can do them two ways, whichever is easier for you.

```
const MediaCheck = ({ media }) => { // { x, y, z }; if you have multiple props
...
```

```
const MediaCheck = (props) => {
   const { media } = props; // { x, y, z } if you have multiple props
...
```
Then, in your code you can use them like this:
```
if(media === 'Github') ...
```

File: src/templates/project/member-media.js [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    if(props.media ==='Github'){
+        return(
+            <a>
+                <FaGithub />
+            </a>
+        )
```
Comment (robchendev) [inline src/templates/project/member-media.js:10]: Tip: For very short JSX return statements here, you can just do this:
```
return <a><FaGithub /></a>
```

File: src/templates/project/project-styling.js [hunk 5]
```diff
@@ -59,4 +59,87 @@ export const Explanation = styled.div`
...
+  span {
+    margin-bottom: 0.1em;
+    // background: black;
+  }
+  margin-bottom: 1.5em;
+  img {
+    width: 150px;
+    height: 150px;
+    border-radius: 75%;
+    border: 2px solid #c3b4fd;
```
Comment (robchendev) [inline src/templates/project/project-styling.js]: If possible, let's move this color to the `src/styles/globalstyles/theme.js` file and give it a descriptive name. You can simply add it to the bottom of the COLORS object like this:
```
lightPurple: {
  light: '#c3b4fd',
  dark: '#c3b4fd',
},
```
and also update `gatsby-ssr.js` by adding this to the bottom of the file, right above the line with `--initial-color-mode`
```
root.style.setProperty(
  '--color-lightPurple',
  colorMode === 'light'
    ? '${COLORS.lightPurple.light}'
    : '${COLORS.lightPurple.dark}'
);
```
then you can write the border css like this:
```
border: 2px solid var(--color-lightPurple, ${COLORS.lightPurple.light});
```
This is so we can easily change the color later and for light theme as well.

File: src/templates/project/member-media.js [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+    return (
+        <h1>ya done goofed</h1>
+    )
+}
+
+const Media = (props) => {
+    return (
+        <MediaCheck media={props.name} />
+    )
+}
```
Comment (robchendev) [inline src/templates/project/member-media.js]: This seems redundant, why not just have `Media` return the entire function scope of `MediaCheck`?

**Review by robchendev — COMMENTED** [review]
LGTM, code style needs some touch-up as mentioned in the code review comments. Look over the review, fix some things to your own preference and upon your next push I'll merge.

File: src/templates/project/member-media.js (hunk 6 above)
Comment (ganmany5288) [reply src/templates/project/member-media.js]: Didn't have that considered lmao

**Review by robchendev — APPROVED** [review]
Very nice work, I'll be making some code-style related changes and maybe some design changes in a later PR.
